### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by Salwa99**
# Blog App - processing data in models

### Hello there! :wave: 

**In this PR I:**

1. Create model classes for all entities, as shown in the ERD diagram.

2. Set up associations between models.

3. Use models to insert data. 

===> **Here are some inserted data:**

Screenshot from 2023-04-20 14-27-47

4. Create custom methods for `User`, `Post`, `Comment`, and `Like`.

5. Follow GitFlow and Ruby best practices.

6. Fix Linter errors.

7. Document my work in a professional way.

Cheers! :heart:

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+  validates :title, :text, presence: true
+
+  def update_user_posts_counter
+    author.posts_counter = author.posts.count
+    author.save
+  end
```
Comment (danmainah) [inline app/models/post.rb:16]: - [ ] [optional] @Salwa99 great work on your update  counter method, however, I would recommend you t use  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_user_posts_counter method every time a new post has been saved to the database, then make the update_user_posts_counter method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_likes_counter
+    post.likes_counter = post.likes.count
+    post.save
+  end
```
Comment (danmainah) [inline app/models/like.rb]: - [ ] [optional] @Salwa99 great work on your update_post_likes_counter method, however, I would recommend you t use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_likes_counter method every time a new post has been saved to the database, then make the update_post_likes_counter method private

```suggestion
  # app/models/post.rb
  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:likes_counter)
  end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_comments_counter
+    post.comments_counter = post.comments.count
+    post.save
+  end
```
Comment (danmainah) [inline app/models/comment.rb:10]: - [ ] [optional] @Salwa99 great work on your update_post_likes_counter method, however, I would recommend you t use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_likes_counter method every time a new post has been saved to the database, then make the update_post_likes_counter method private

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Salwa99 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️

- [x] @Salwa99 Great work on your documentation, however, the link to your issues page is broken, check how you reference the page (../../../issues), `hint remove one ../`
image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Salwa99** [comment]
Thank you @danmainah, I'll implement the optional changes.

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Salwa99, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Blog data models implemented correctly :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive summary and title :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:    

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**